packages/php/email-editor/src/Engine/Renderer/ContentRenderer/class-content-renderer.php

```diff
@@ -161,8 +172,21 @@ public function render_block( string $block_content, array $parsed_block ): stri
 		$context = new Rendering_Context( $this->theme_controller->get_theme() );
 
 		$block_type = $this->block_type_registry->get_registered( $parsed_block['blockName'] );
-		if ( $block_type && isset( $block_type->render_email_callback ) && is_callable( $block_type->render_email_callback ) ) {
-			return call_user_func( $block_type->render_email_callback, $block_content, $parsed_block, $context );
+		try {
+			if ( $block_type && isset( $block_type->render_email_callback ) && is_callable( $block_type->render_email_callback ) ) {
+				return call_user_func( $block_type->render_email_callback, $block_content, $parsed_block, $context );
+			}
+		} catch ( \Exception $error ) {
+			$this->logger->error(
+				'Error thrown while rendering block.',
+				array(
+					'exception'    => $error,
+					'block_name'   => $parsed_block['blockName'],
+					'parsed_block' => $parsed_block,
+					'message'      => $error->getMessage(),
+				)
+			);
+			exit();
```

Comment:
Calling `exit();` feels a bit too strong here. Imagine generating multiple emails. What about returning an empty string or the $block_content here?

Reply:
This is an error because I was testing the logging. I will fix it.